### DISCUSSION TITLE
.npmrc does not work in private repos

### DISCUSSION THREAD

**Discussion by TrejGun** [renovatebot/renovate#11498] [Request Help]
Hi guys! 
This discussion is a continuation of the issue https://github.com/renovatebot/renovate/issues/11300

We have a setup in our GH organization with several private repositories which depends one on another and a repo with renovate config. We also have public repositories and use both types of dependencies in our package.json. Each of the repositories with code has .npmrc with GH token and this worked perfectly fine until recently. But about a week ago we realized this does not work anymore so we have to encrypt the token and put it into a shared renovate config. It looks like this was an unintended side effect so please check and confirm what has happened

we have prepared a minimal reproduction example with two repos but we can't make them public because GH will cancel the token, so we have to invite you as collaborators. Please give us logins so we can do this. 

@rarkins  @viceice

**Comment by viceice** [comment]
So yes, this is intended behavior. It bad practice to have plain token in git repo anyways.

So your solution is to encrypt the token and add to a hostrule.

```json
{
  "hostRules": [
    {
      "matchHost": "https://npm.pkg.github.com/",
      "hostType": "npm",
      "encrypted": {
        "token": "<Encrypted PAT Token>"
      }
    }
  ],
  "npmrc": "@organizationName:registry=https://npm.pkg.github.com/"
}
```

https://docs.renovatebot.com/getting-started/private-packages/#automatically-authenticate-for-npm-package-stored-in-private-github-npm-repository

**Comment by TrejGun** [comment]
thanks
